src/fu-engine.c

```diff
@@ -3485,6 +3486,16 @@ fu_engine_load_plugins (FuEngine *self, GError **error)
 			}
 		}
 
+		/* plugin does not match built version */
+		if (fu_plugin_get_built_version (plugin) == NULL) {
+			g_warning ("plugin %s should call fu_plugin_init_real()", name);
+		} else if (g_strcmp0 (fu_plugin_get_built_version (plugin),
+				      FWUPD_BUILT_VERSION) != 0) {
+			g_warning ("plugin %s has built version %s, not %s",
```

Comment:
I think that this information should get saved further into the engine too.

My thought:
* It should actually be something that a client can check.  Proposal: `fwupd_client_daemon_tainted`
* During startup of any fwupd frontend that supports this check, if tainted is false, show a warning in the frontend.

Reply:
Agreed, I'll add this.